src/openrct2/park/ParkFile.cpp

```diff
@@ -2312,6 +2313,31 @@ int32_t scenario_save(u8string_view path, int32_t flags)
     catch (const std::exception& e)
     {
         log_error(e.what());
+
+        Formatter ft;
+        ft.Add<const char*>(e.what());
+        context_show_error(STR_FILE_DIALOG_TITLE_SAVE_SCENARIO, STR_STRING, ft);
+        gfx_invalidate_screen();
+
+        auto ctx = OpenRCT2::GetContext();
+        auto uictx = ctx->GetUiContext();
+
+        std::string title = "Error while saving";
+        std::string message
+            = "There was an error while saving scenario.\nhttps://github.com/OpenRCT2/OpenRCT2/issues/17664\nWe would like to "
+              "collect more information about this issue, if this did not happen due to missing permissions, lack of space, "
+              "etc. please consider submitting a bug report. To collect information we would like to trigger an assert.";
```

Comment:
Users may click trigger an assert without knowing the consequences which I assume will be losing their saved game. This is not great if it was a trivial issue to why it couldn't save, like invalid permissions.

The issue number is temporary, don't we want to keep this message forever - in which case it should be more generalised?

Reply:
I don't intend this code to stay in the game forever - as we fix the original issue, I hope this to go away. They will also _not_ be losing their park any more than they already have - in case of crash we do attempt to save it anyway (not via scenario_save though, so shouldn't retrigger that error anymore).

Reply:
> They will also not be losing their park any more than they already have

I am talking about if it was just a simple issue like permission denied. The user might get confused and click the assert button. If the crash fails to save it then they would lose their work. I am just suggesting the message say that the assert button will terminate the game.

Reply:
I already explain those in the error message. What `assert` does is entirely implementation-dependent